src/test/javascript/spec/component/course/course-overview.component.spec.ts

```diff
@@ -0,0 +1,257 @@
+                MockDirective(MockHasAnyAuthorityDirective),
+                MockDirective(OrionFilterDirective),
+                MockPipe(TranslatePipe),
+                MockDirective(JhiSortDirective),
+                MockDirective(JhiSortByDirective),
+                MockPipe(ArtemisDatePipe),
+                MockComponent(CourseExerciseRowComponent),
+                MockComponent(CourseExercisesComponent),
+                MockComponent(CourseRegistrationSelectorComponent),
+                MockComponent(CourseCardComponent),
+                MockComponent(SecuredImageComponent),
+                MockComponent(AlertComponent),
+                MockComponent(AlertErrorComponent),
+            ],
+            providers: [
+                { provide: LocalStorageService, useClass: MockSyncStorage },
+                { provide: SessionStorageService, useClass: MockSyncStorage },
+                { provide: TranslateService, useClass: MockTranslateService },
+                { provide: ActivatedRoute, useValue: route },
+                { provide: CourseExerciseRowComponent },
+                { provide: JhiAlertService, useClass: MockAlertService },
+                { provide: Router, useClass: MockRouter },
+            ],
+        })
+            .compileComponents()
+            .then(() => {
+                fixture = TestBed.createComponent(CourseOverviewComponent);
+                component = fixture.componentInstance;
+                courseService = TestBed.inject(CourseManagementService);
+                courseScoreCalculationService = TestBed.inject(CourseScoreCalculationService);
+                teamService = TestBed.inject(TeamService);
+                jhiWebsocketService = TestBed.inject(JhiWebsocketService);
+                fixture.detectChanges();
+            });
+    });
+
+    afterEach(() => {
+        component.ngOnDestroy();
+        sinon.restore();
+    });
+
+    it('Should call all methods on init', fakeAsync(() => {
+        const getCourseStub = stub(courseScoreCalculationService, 'getCourse');
+        const subscribeToTeamAssignmentUpdatesStub = stub(component, 'subscribeToTeamAssignmentUpdates');
+        const subscribeForQuizChangesStub = stub(component, 'subscribeForQuizChanges');
+        const adjustCourseDescriptionStub = stub(component, 'adjustCourseDescription');
+        const findOneForDashboardStub = stub(courseService, 'findOneForDashboard');
+        findOneForDashboardStub.returns(of(new HttpResponse({ body: course1, headers: new HttpHeaders() })));
+        getCourseStub.returns(course1);
+
+        component.ngOnInit();
+        tick(1000);
+
+        expect(getCourseStub).to.have.been.called;
+        expect(adjustCourseDescriptionStub).to.have.been.called;
+        expect(subscribeForQuizChangesStub).to.have.been.called;
+        expect(subscribeToTeamAssignmentUpdatesStub).to.have.been.called;
+    }));
+
+    it('Should call load Course methods on init', fakeAsync(() => {
```

Comment:
I don't understand this test case, isn't it exactly the same like the one above just without `getCourseStub.returns(course1);`? (Which we don't test for anyway?)